packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -99,6 +99,7 @@ export const SelectPopover = styled( Ariakit.SelectPopover )`
 	z-index: 9999; // Ensure the popover is on top
 	position: absolute !important;
 	max-height: 400px;
+	overflow: auto;
```

Comment:
This works in synergy with https://github.com/WordPress/gutenberg/pull/62424/commits/a1b9b3e9b18e93c5e24ecb40a7f9e6bbda7f296d, and allows the user to scroll up/down when hovering over the popover and the size of items is > the max-height of the container.

Reply:
Opened https://github.com/WordPress/gutenberg/pull/62821 to work on the select popover fixes separately